src/test/java/teammates/test/pageobjects/AppPage.java

```diff
@@ -903,6 +918,20 @@ public AppPage verifyHtmlMainContent(String filePath) throws IOException {
         return verifyHtmlPart(MAIN_CONTENT, filePath);
     }
 
+    public AppPage verifyHtmlMainContentWithReloadRetry(final String filePath) throws IOException {
+        return RetryManager.runUntilNoException(new RetryableTaskReturnsThrows<AppPage, IOException>("HTML verification") {
+            @Override
+            public AppPage run() throws IOException {
```

Comment:
The task is supposed to halt immediately and throw the exception upwards (this is the original behaviour). IOException will only happen if the wrong filename was specified (or other such errors out of our control), so no point retrying. The purpose of the Throws parameter is to indicate what kind of exception should immediately halt the task and be thrown upwards.